packages/types/index.ts

```diff
@@ -0,0 +1,161 @@
+import type { AsyncIterator } from 'asynciterator';
+import type { Map } from 'immutable';
+import type * as RDF from 'rdf-js';
+import type { Algebra } from 'sparqlalgebrajs';
+
+/**
+ * An immutable solution mapping object.
+ * This maps variables to a terms.
+ *
+ * Variables are represented as strings containing the variable name prefixed with '?'.
+ * Blank nodes are represented as strings containing the blank node name prefixed with '_:'.
+ * Terms are named nodes, literals or the default graph.
+ */
+export type IBindings = Map<string, RDF.Term>;
```

Comment:
Any reason why this is prefixed with `I`, while the old type wasn't?
(Also applicable for other types in this file)

If no particular reason, I'd keep the type names exactly the same as before, as to not confuse users.

Reply:
I think `IBindings` is a clearer indicator of what that is - a TS typing - and I think the `@comunica/types` could benefit from the additional clarity (given that most other types and interfaces do start with `I`). Same applies to `IBindingsStream`.

Reply:
We currently follow the convention that `interface`'s start with `I`, and types don't start with `I`.
I'm open to changing this, but then we'd have to make this consistent across the whole codebase, and make this a breaking change.

Reply:
Ah, you're right! My bad. Will drop the `I` from types. Apologies for the latency, I'm currently on a business trip and have very little time for anything else but the tests we're carrying out. At least the facility is nice!

Reply:
No worries, there's no hurry on my end. Have fun there!

Reply:
See my comments above. I've replaced the `I` with `T` in the `@comunica/types` package as that felt like the most coherent solution but kept the original names when re-exporting types in packages like `@comunica/bus-query-operation` to maintain backward compatibility.

Reply:
I guess you can resolve this as we're also discussing the renaming issue in [the thread above](https://github.com/comunica/comunica/pull/784#discussion_r587715184).